packages/console/src/utilities/validator.ts

```diff
@@ -1,12 +1,16 @@
-export const uriValidator = (value: string) => {
+import { ApplicationType } from '@logto/schemas';
+import { redirectNativeRegEx, redirectUriRegEx } from '@logto/shared';
+
+export const uriValidator = (value: string, applicationType?: ApplicationType) => {
   try {
-    // eslint-disable-next-line no-new
-    new URL(value);
+    if (applicationType === ApplicationType.Native) {
+      return redirectNativeRegEx.test(value);
+    }
+
+    return redirectUriRegEx.test(value);
   } catch {
     return false;
   }
```

Comment:
If we decide not to use `new URL(value)`, we can remove the try-catch clause.
Or you can keep the try-catch, and combine the usages of  `new URL` and `regex.test()`. Either is fine to me